lib/ui/window.dart

```diff
@@ -5,614 +5,72 @@
+/// without gaps.
 ///
-/// The [Window.padding] property is computed from both [Window.viewInsets] and
-/// [Window.viewPadding]. It will allow a view inset to consume view padding
-/// where appropriate, such as when a phone's keyboard is covering the bottom
-/// view padding and so "absorbs" it.
+/// The [padding] property is computed from both
+/// [viewInsets] and [viewPadding]. It will allow a
+/// view inset to consume view padding where appropriate, such as when a phone's
+/// keyboard is covering the bottom view padding and so "absorbs" it.
 ///
 /// Clients that want to position elements relative to the view padding
-/// regardless of the view insets should use the [Window.viewPadding] property,
-/// e.g. if you wish to draw a widget at the center of the screen with respect
-/// to the iPhone "safe area" regardless of whether the keyboard is showing.
+/// regardless of the view insets should use the [viewPadding]
+/// property, e.g. if you wish to draw a widget at the center of the screen with
+/// respect to the iPhone "safe area" regardless of whether the keyboard is
+/// showing.
 ///
-/// [Window.padding] is useful for clients that want to know how much padding
-/// should be accounted for without concern for the current inset(s) state, e.g.
-/// determining whether a gesture should be considered for scrolling purposes.
-/// This value varies based on the current state of the insets. For example, a
-/// visible keyboard will consume all gestures in the bottom part of the
-/// [Window.viewPadding] anyway, so there is no need to account for that in the
-/// [Window.padding], which is always safe to use for such calculations.
-class Window {
-  Window._() {
-    _setNeedsReportTimings = _nativeSetNeedsReportTimings;
-  }
+/// [padding] is useful for clients that want to know how much
+/// padding should be accounted for without concern for the current inset(s)
+/// state, e.g. determining whether a gesture should be considered for scrolling
+/// purposes. This value varies based on the current state of the insets. For
+/// example, a visible keyboard will consume all gestures in the bottom part of
+/// the [viewPadding] anyway, so there is no need to account for
+/// that in the [padding], which is always safe to use for such
+/// calculations.
+///
+/// See also:
+///
+///  * [FlutterWindow], a special case of a [FlutterView] that is represented on
+///    the platform as a separate window which can host other [FlutterView]s.
+abstract class FlutterView {
+  /// The platform dispatcher that this view is registered with, and gets its
+  /// information from.
+  PlatformDispatcher get platformDispatcher;
 
-  /// The number of device pixels for each logical pixel. This number might not
-  /// be a power of two. Indeed, it might not even be an integer. For example,
-  /// the Nexus 6 has a device pixel ratio of 3.5.
+  /// The configuration of this view.
+  ViewConfiguration get viewConfiguration;
```

Comment:
If everything in the view has convenience getters on FlutterView - does this need to be public?

Reply:
Or is this only public so subclasses can override it since we can't mark it `@protected` here?